src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1116,32 +1116,27 @@ void OrbitMainWindow::LoadCaptureOptionsIntoApp() {
     app_->SetSamplesPerSecond(0.0);
   }
 
-  if (!app_->IsDevMode()) {
-    app_->SetUnwindingMethod(orbit_qt::CaptureOptionsDialog::kCallstackUnwindingMethodDefaultValue);
-    app_->SetStackDumpSize(std::numeric_limits<uint16_t>::max());
-  } else {
-    UnwindingMethod unwinding_method = static_cast<UnwindingMethod>(
+  UnwindingMethod unwinding_method = static_cast<UnwindingMethod>(
+      settings
+          .value(kCallstackUnwindingMethodSettingKey,
+                 static_cast<int>(
+                     orbit_qt::CaptureOptionsDialog::kCallstackUnwindingMethodDefaultValue))
+          .toInt());
+  if (unwinding_method == CaptureOptions::kUndefined) {
```

Comment:
This is existing code, and maybe I'm missing some surrounding code that would prevent this, but if the cast is from an invalid value (corrupted settings?), we have no guarantee `unwinding_method` is one of the three possible options (it's undefined behavior and I guess it will contain the invalid value). I think we should compare `unwinding_method` to the two valid values instead.

Reply:
Makes sense. Done.